happypose/toolbox/utils/download.py

```diff
@@ -173,27 +180,135 @@ def main():
         )
         for result_id in (PBR_INFERENCE_ID, SYNT_REAL_INFERENCE_ID, SYNT_REAL_ICP_INFERENCE_ID,
                           SYNT_REAL_4VIEWS_INFERENCE_ID, SYNT_REAL_8VIEWS_INFERENCE_ID):
-            download(f'cosypose/results/{result_id}', LOCAL_DATA_DIR / 'results')
+            download(f'{DOWNLOAD_URL}/cosypose/results/{result_id}', LOCAL_DATA_DIR / 'results')
 
 
-def run_rclone(cmd, args, flags):
-    rclone_cmd = ['rclone', cmd] + args + flags + ['--config', str(RCLONE_CFG_PATH)]
-    logger.debug(' '.join(rclone_cmd))
-    print(rclone_cmd)
-    subprocess.run(rclone_cmd)
+#manages multiple downloads in async
+def downloads(*args):
+    asyncio.run(adownloads(*args))
 
+async def adownloads(*args):
+    async with DownloadClient() as dl_client:
+        for (download_path, local_path,new_flags) in args:
+            if flags is None:
+                flags = []
+            flags = Flags(new_flags)
+            dl_client.create_task(dl_client.adownload(download_path, local_path,flags))
 
-def download(download_path, local_path, flags=[]):
-    download_path = Path(download_path)
-    if download_path.name != local_path.name:
-        local_path = local_path / download_path.name
-    if '.' in str(download_path):
-        rclone_path = RCLONE_ROOT + str(download_path)
-    else:
-        rclone_path = RCLONE_ROOT + str(download_path) + "/"
-    local_path = str(local_path)
-    logger.info(f"Copying {rclone_path} to {local_path}")
-    run_rclone("copyto", [rclone_path, local_path], flags=flags + ["-P"])
+
+def download(download_path, local_path, flags=None):
+    if flags is None:
+        flags = []
+    flags = Flags(flags)
+    asyncio.run(DownloadClient.run(download_path, local_path, flags))
+
+class DownloadClient:
+    def __init__(self):
+        self.client = httpx.AsyncClient()
+        self.task_set = set()
+    
+    async def __aenter__(self):
+        return self
+
+    async def __aexit__(self, *args):
+        await self.aclose()
+    
+    async def aclose(self):
+        while len(self.task_set) > 0:
+            for task in list(self.task_set):
+                await task
```

Comment:
It looks like this is a job for [`asyncio.gather`](https://docs.python.org/3.11/library/asyncio-task.html#asyncio.gather), could you try with it ?